New script - add to core party (#1146)
DFHack/scripts: docs/advtools.rst

.. dfhack-tool::
    :summary: A collection of useful adventure mode tools.
    :tags: adventure interface gameplay units

Usage
-----

::

    advtools party

Examples
--------

``advtools party``
    Shows a dialog prompt to promote your extra party members to your core (controllable) party.

``party`` Command
-----------------

> When you run this command, it will show you a list of your extra party and let you choose

Reviewer: ```suggestion
When you run this command, you will get a list of your extra party members and can choose
```
